readthedocs/projects/models.py

```diff
@@ -350,12 +350,26 @@ class Project(models.Model):
     conf_py_file = models.CharField(
         _('Python configuration file'),
         max_length=255,
-        default='',
+        default="",
         blank=True,
         help_text=_(
-            'Path from project root to <code>conf.py</code> file '
-            '(ex. <code>docs/conf.py</code>). '
-            'Leave blank if you want us to find it for you.',
+            "Path from project root to <code>conf.py</code> file "
+            "(ex. <code>docs/conf.py</code>). "
+            "Leave blank if you want us to find it for you.",
+        ),
+    )
+
+    build_config_file = models.CharField(
+        _("Build configuration file"),
+        max_length=1024,
+        default=None,
+        blank=True,
+        null=True,
+        help_text=_(
+            "<strong>Warning: experimental feature</strong>. "
+            "Custom path from repository root to a build configuration file, "
+            "ex. <code>subproject/docs/.readthedocs.yaml</code>. "
```

Comment:
```suggestion
            "ex. <code>subpath/docs/.readthedocs.yaml</code>. "
```

We don't want to overload `subproject` if we can avoid it.